cloudify_cli/commands/plugins.py

```diff
@@ -215,6 +221,93 @@ def get(plugin_id, logger, client, tenant_name, get_data):
     print_single(columns, plugin, 'Plugin:')
 
 
+def _wait_for_plugin_to_be_installed(client, plugin_id, managers, agents,
+                                     timeout, logger):
+    logger.info(
+        'Waiting for plugin %s to be installed on the managers: [%s] '
+        'and agents: [%s]',
+        plugin_id, ', '.join(managers), ', '.join(agents)
+    )
+    wait_managers = set(managers)
+    wait_agents = set(agents)
+    errors = 0
+    deadline = time.time() + timeout
+    while time.time() < deadline:
+        for pstate in client.plugins.get(plugin_id)['installation_state']:
+
+            if pstate['state'] == PluginInstallationState.INSTALLED:
+                if pstate.get('manager') in wait_managers:
+                    wait_managers.remove(pstate['manager'])
+                    logger.info('Finished installing on manager %s',
+                                pstate['manager'])
+                if pstate.get('agent') in wait_agents:
+                    wait_agents.remove(pstate['agent'])
+                    logger.info('Finished installing on agent %s',
+                                pstate['agent'])
+
+            if pstate['state'] == PluginInstallationState.ERROR:
+                if pstate.get('manager') in wait_managers:
+                    errors += 1
+                    wait_managers.remove(pstate['manager'])
+                    logger.info('Error installing on manager %s: %s',
+                                pstate['manager'], pstate['error'])
+                if pstate.get('agent') in wait_agents:
+                    errors += 1
+                    wait_agents.remove(pstate['agent'])
+                    logger.info('Error installing on agent %s: %s',
+                                pstate['agent'], pstate['error'])
```

Comment:
I'm not turbo happy with this code but I haven't been able to structure it better :(